Lib/test/test_subprocess.py

```diff
@@ -1824,6 +1824,22 @@ def test_encoding_warning(self):
         self.assertTrue(lines[0].startswith(b"<string>:2: EncodingWarning: "))
         self.assertTrue(lines[2].startswith(b"<string>:3: EncodingWarning: "))
 
+    def test_shell(self):
+        # Test basic echo command
+        proc = subprocess.shell("echo Python", stdout=subprocess.PIPE, text=True)
+        self.assertEqual(proc.returncode, 0)
+        self.assertEqual(proc.stdout.rstrip(), "Python")
+
+        # cmd type must be str
+        with self.assertRaises(TypeError):
+            subprocess.shell(b"echo Python")
+        with self.assertRaises(TypeError):
+            subprocess.shell(["echo", "Python"])
+
+        # Passing shell=False is invalid
+        with self.assertRaises(TypeError):
```

Comment:
Do you think it's better to have a TypeError or rather a ValueError? The TypeError would tell you "got multiple values for keyword argument 'shell'" but I think it's preferrable for the user to know that they are *not* allowed at all to pass 'shell=False' instead.

Reply:
TypeError is a standard error when you pass a keyword argument twice.

```
>>> def f(**kwargs): pass
... 
>>> f(x=1, **{'x': 1})
TypeError: __main__.f() got multiple values for keyword argument 'x'
```

Reply:
Ah maybe I wasn't clear enough. From a user perspective, I would expect the error to tell me that I am not allowed to pass `shell=False` because of the interface itself, and not because it's indirectly using `shell=True` before.

I would have expected something like:

"This function does not support `shell=False`" instead of the cryptic "got multiple values for keyword argument 'shell'" (if you use `subprocess.shell(*args, **kwargs)`) then you could wonder "but why is it telling me that I passed the keyword multiple times when I did not?")

Reply:
TypeError is also the error type is you pass an argument which doesn't exist to a function:

```
>>> def f(): passs
... 
>>> f(shell=True)
TypeError: f() got an unexpected keyword argument 'shell'
```

I agree that the error message is not great, but IMO TypeError is fine.

Reply:
Actually, what I would have done is:

```python
def shell(cmd, /, **kwargs): 
	if not kwargs.setdefault('shell', True):
		raise ValueError("the 'shell' argument must be True or unspecified")
	...
```

Anyway, maybe you should add in the docs that this function must *not* be called at all with the `shell` keyword? by the way, do you think `cmd` should be positional only? (I don't know the standard library likes using positional-only arguments)

Reply:
Done